provisioning/roles/php/tasks/main.yml

```diff
@@ -1,22 +1,43 @@
 - set_fact: php_version_installed={{ php_version | float }}
+  
+- fail: msg="PHP7.0 is only available on Jessie."
+  when: "ansible_lsb.major_release|int < 8 and {{ php_version_installed }} == 7.0"
+
+- include: ubuntu-repo.yml
+  when: "ansible_distribution == 'Ubuntu'"
 
-- name: Install DotDeb key
-  apt_key: url=http://www.dotdeb.org/dotdeb.gpg
+- include: debian-repo.yml
+  when: "ansible_distribution == 'Debian'"
+
+- include_vars: "{{ item }}"
+  with_first_found:
+   - "{{ ansible_distribution }}-{{ php_version_installed }}.yml"
+   - "{{ ansible_distribution }}.yml"
+
+- name: install PHP packages
+  apt: pkg={{ item }} state=latest
   sudo: yes
-  when: "{{ php_version_installed }} > 5.4"
+  with_items: "{{ php_packages }}" 
 
-- fail: msg="PHP7.0 is only available on Jessie."
-  when: "ansible_lsb.major_release|int < 8 and {{ php_version_installed }} == 7.0"
+- name: install mysql database driver
+  apt: pkg={{ php_mysql_package }} state=latest
+  sudo: yes
+  when: "'{{ database_type|default(false) }}' == 'mysql'"
+
+- name: install postgresql database driver
+  apt: pkg={{ php_pgsql_package}} state=latest
+  sudo: yes
+  when: "'{{ database_type|default(false) }}' == 'postgresql'"
+
+- include: not-debian-7.0.yml
+  when: "not ({{ php_version_installed }} == 7.0 and ansible_distribution == 'Debian')"
 
-- include: php7.yml
-  when: "{{ php_version_installed }} == 7.0"
+- include: debian-php7.yml
```

Comment:
because we have to compile xdebug and imagick on Debian php7 for now. There doesn't seem to be a package yet for those. Once there are, this can be removed